Tests/RockLib.Messaging.SQS.Tests/SQSTests.cs

```diff
@@ -412,6 +412,54 @@ public static void SQSReceiverMessageWithUnpackSNSTrueCorrectlyUnpacksSNSMessage
             sqsReceiverMessage.Headers["TopicARN"].Should().Be("arn:PutARealARNHere");
         }
 
+        [Fact]
+        public static void SQSReceiverMessageHasSQSMessageIDSetCorrectly()
+        {
+            var message = new Message
+            {
+                MessageId = (new Guid()).ToString(),
+                Body = @"{""TestMessageItem"":""ThisIsAFakeItem""}"
+            };
+
+            var sqsReceiverMessage = new SQSReceiverMessage(message, c => Task.FromResult(0), c => Task.FromResult(0), unpackSNS: false);
+
+            var messageId = sqsReceiverMessage.Headers.GetValue<string>("SQS.MessageID");
+
+            messageId.Should().NotBeNull();
+
+            sqsReceiverMessage.Dispose();
+        }
+
+        [Fact]
+        public static void SQSReceiverMessageDoesNotHaveSQSMessageIDHeaderWhenSNS()
+        {
+
+            var snsMessage = @"{
+  ""Type"" : ""Notification"",
+  ""MessageId"" : ""f5129dcc-0bb0-5e18-9431-c95b6a9b32d9"",
+  ""TopicArn"" : ""arn:PutARealARNHere"",
+  ""Message"" : ""This is a better test message"",
+  ""Timestamp"" : ""2018-12-21T21:45:15.114Z"",
+  ""SignatureVersion"" : ""1"",
+  ""Signature"" : ""SomeSignatureValue"",
+  ""SigningCertURL"" : ""SomeUrl"",
+  ""UnsubscribeURL"" : ""SomeOtherUrl""
+}";
+
+            var message = new Message
+            {
+                MessageId = (new Guid()).ToString(),
+                Body = snsMessage
+            };
+
+            var sqsReceiverMessage = new SQSReceiverMessage(message, c => Task.FromResult(0), c => Task.FromResult(0), unpackSNS: true);
+ 
+            Assert.Throws<KeyNotFoundException>(() => sqsReceiverMessage.Headers.GetValue<string>("SQS.MessageID"));
+
+            sqsReceiverMessage.Dispose();
```

Comment:
Same here.  Add the `using` statement and get rid of the `sqsReceiverMessage.Dispose()`.